### PR TITLE
Add provisioning logic

### PR DESCRIPTION
**Pull request by dhellmann**
This successfully triggers provisioning.

### CODE REVIEW

File: deploy/crds/worker-0.yaml [hunk 1]
```diff
@@ -19,3 +19,6 @@ spec:
     address: libvirt://192.168.122.1:6233/
     credentialsName: worker-0-bmc-secret
   bootMACAddress: 00:73:49:3a:76:8e
+  image:
+    url: "http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2"
+    checksum: "97830b21ed272a3d854615beb54cf004"
```
Comment (dtantsur) [inline deploy/crds/worker-0.yaml]: btw we understand URLs here starting with Stein IPA images.

File: pkg/provisioner/ironic/ironic.go [hunk 2]
```diff
@@ -390,6 +383,165 @@ func (p *ironicProvisioner) InspectHardware() (result provisioner.Result, err er
...
+		}
+		p.status.State = stateMakingAvailable
+		result.Dirty = true
+		return result, nil
+	}
+
+	// Wait for the host to become available
+	if p.status.State == stateMakingAvailable {
+		if ironicNode.ProvisionState != nodes.Available {
+			p.log.Info("waiting for host to become available")
```
Comment (stbenjam) [inline pkg/provisioner/ironic/ironic.go]: What happens if a host never becomes available? For example, Ironic goes to `clean failed` - does this mean it'll just return dirty forever?

Comment (dhellmann) [reply pkg/provisioner/ironic/ironic.go]: Yeah, for this first pass I'm just trying to get 1 happy path working. There are going to be lots of error handling updates to make.

**Comment by dhellmann** [comment]
I rebased this. The latest series of patches allows me to write the image to the VM, and the VM boots. I would like to merge this before working on some of the error handling cases.

**Comment by russellb** [comment]
I'm fine with this based on existing reviews.  @dhellmann I'll leave the merge to you so you can decide how you want to squash this

**Comment by dhellmann** [comment]
I rebased and squashed a couple of the fix-up patches into the main provisioning patch, but I'll leave the others as they are because they're somewhat unrelated changes.

**Pull request merged**